internal/api/pkce.go

```diff
@@ -70,7 +69,9 @@ func validatePKCEParams(codeChallengeMethod, codeChallenge string) error {
 	case codeChallenge != "" && codeChallengeMethod == "":
 		return badRequestError(InvalidPKCEParamsErrorMessage)
 	case codeChallenge != "" && codeChallengeMethod != "":
-		break
+		if valid, err := isValidCodeChallenge(codeChallenge); !valid {
+			return err
+		}
```

Comment:
can we combine these 2 cases since the difference is just whether the `codeChallengeMethod` is empty?

Reply:
will file a FLUP in a few hours or tmrw

Reply:
Actually,  I came back to re-read this and was a little confused - which two cases did you have in mind? 

Would it be:
```
case codeChallenge != "" && codeChallengeMethod == "":
       return badRequestError(InvalidPKCEParamsErrorMessage)
case codeChallenge != "" && codeChallengeMethod != "":
       if valid, err := isValidCodeChallenge(codeChallenge); !valid {
           return err
       }
```

Reply:
yeah wanted to combine it coz it looks similar and only differs by 1 char:

```
case codeChallenge != "":
       if  codeChallengeMethod == "" {
           return badRequestError(InvalidPKCEParamsErrorMessage)
       } else {
           if valid, err := isValidCodeChallenge(codeChallenge); !valid {
               return err
           }
       }
```